nimble/host/services/audio/bass/src/ble_audio_svc_bass.c

```diff
@@ -0,0 +1,915 @@
+{
+    uint16_t uuid16 = ble_uuid_u16(ctxt->chr->uuid);
+    int rc;
+
+    switch (uuid16) {
+    case BLE_SVC_AUDIO_BASS_CHR_UUID16_BAS_CONTROL_POINT:
+        if (ctxt->op == BLE_GATT_ACCESS_OP_WRITE_CHR) {
+            rc = ble_svc_audio_bass_ctrl_point_write_access(ctxt);
+        } else {
+            assert(0);
+        }
+        return rc;
+    case BLE_SVC_AUDIO_BASS_CHR_UUID16_BROADCAST_RECEIVE_STATE:
+        if (ctxt->op == BLE_GATT_ACCESS_OP_READ_CHR) {
+            rc = ble_svc_audio_bass_rcv_state_read_access(ctxt, arg);
+        } else {
+            assert(0);
+        }
+        return rc;
+    default:
+        assert(0);
+    }
+}
+
+static uint8_t
+ble_svc_audio_bass_get_new_source_id()
+{
+    struct ble_svc_audio_bass_receiver_state *rcv_state;
+    struct ble_svc_audio_bass_receiver_state *next_rcv_state;
+    uint8_t free_source_id;
+    uint8_t highest_source_id = 0;
+    uint8_t lowest_source_id = BLE_SVC_AUDIO_BASS_RECEIVE_STATE_SRC_ID_NONE;
+
+    int i;
+
+    /* First characteristic is BLE_SVC_AUDIO_BASS_CHR_UUID16_BAS_CONTROL_POINT, iterate from 2nd */
+    for (i = 1; i <= MYNEWT_VAL(BLE_SVC_AUDIO_BASS_RECEIVE_STATE_MAX); i++) {
+        rcv_state = ble_svc_audio_bass_chrs[i].arg;
+        if (highest_source_id < rcv_state->source_id &&
+            rcv_state->source_id != BLE_SVC_AUDIO_BASS_RECEIVE_STATE_SRC_ID_NONE)
+        {
+            highest_source_id = rcv_state->source_id;
+        }
+
+        if (lowest_source_id > rcv_state->source_id) {
+            lowest_source_id = rcv_state->source_id;
+        }
+    }
+
+    if ((highest_source_id + 1) < BLE_SVC_AUDIO_BASS_RECEIVE_STATE_SRC_ID_NONE) {
+        return highest_source_id + 1;
+    }
+
+    return min(lowest_source_id + 1, 0);
+}
+
+static int
+ble_svc_audio_bass_remote_scan_stopped(uint8_t *data, void *arg)
+{
+    struct ble_svc_audio_bass_ctrl_point_event ev = {
+        .op = BLE_SVC_AUDIO_BASS_CTRL_POINT_EVENT_REMOTE_SCAN_STOPPED
+    };
+
+    if (ctrl_point_ev.ctrl_point_ev_fn) {
+        ctrl_point_ev.ctrl_point_ev_fn(&ev, ctrl_point_ev.arg);
+    }
+
+    return 0;
+}
+
+static int
+ble_svc_audio_bass_remote_scan_started(uint8_t *data, void *arg)
+{
+    struct ble_svc_audio_bass_ctrl_point_event ev = {
+        .op = BLE_SVC_AUDIO_BASS_CTRL_POINT_EVENT_REMOTE_SCAN_STARTED
+    };
+
+    if (ctrl_point_ev.ctrl_point_ev_fn) {
+        ctrl_point_ev.ctrl_point_ev_fn(&ev, ctrl_point_ev.arg);
+    }
+
+    return 0;
+}
+
+static int
+ble_svc_audio_bass_create_bsnk(ble_addr_t *addr, uint8_t sid,
+                               uint32_t broadcast_id, uint16_t pa_itvl,
+                               struct ble_audio_bsnk **out_bsnk)
+{
+    struct ble_audio_bsnk_create_params bsnk_create_params;
+
+    bsnk_create_params.adv_addr = addr;
+    bsnk_create_params.adv_sid = sid;
+    bsnk_create_params.broadcast_id = broadcast_id;
+    bsnk_create_params.pa_interval = pa_itvl;
+
+    return ble_audio_bsnk_create(&bsnk_create_params, out_bsnk);
```

Comment:
This should be an event, since Scan Delegator can be standalone proxy.
3.1. BAP role support requirements BAP 1.0 states that Broadcast Sink is dependent on Scan Delegator.